File: src/demo/led_display.ts
import * as gpio from 'rpio';
import { setTimeout } from 'timers';

import LedMatrix from '../display/LedMatrix';
import TorusScreen from '../screen/TorusScreen';
import numbers from '../shapes/numbers';

async function sleep(wait: number) {
  return new Promise((resolve) => setTimeout(resolve, wait));
}

const ROWS = [8, 10, 12, 16, 18, 22, 24, 26];
const COLS = [19, 21, 23, 29, 31, 33, 35, 37].reverse();

const display = new LedMatrix(ROWS, COLS);

const screen = new TorusScreen(8, 8, [numbers[0]]);

display.show(screen).then(() => console.log('done showing'));

sleep(2000).then(() => {
  console.log('done sleeping');
  display.clear();
});

// display.clear();
// display.tearDown();

File: led_display.py
from time import sleep, time
import RPi.GPIO as GPIO

from src.display.LedMatrix import LedMatrix
from src.screen.TorusScreen import TorusScreen
from src.shapes.big import shapes
from src.shapes.SimpleShape import SimpleShape

GPIO.setmode(GPIO.BOARD)

rows = [8, 10, 12, 16, 18, 22, 24, 26]
cols = list(reversed([19, 21, 23, 29, 31, 33, 35, 37]))

display = LedMatrix(rows, cols)

screen = TorusScreen(8, 8, [SimpleShape(shapes['hearts'])])

try:
  display.clear()

  for symbol, shape in shapes.items():
    display.show(TorusScreen(8, 8, [SimpleShape(shape)]))
    sleep(.2)
finally:
  display.tearDown()
